.devcontainer/devcontainer.json

```diff
@@ -1,55 +1,30 @@
-// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
-// https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/python-3-anaconda
+// For format details, see https://aka.ms/devcontainer.json
 {
-	"name": "Numpy (devcontainer)",
-	"build": { 
-		"context": "..",
-		"dockerfile": "Dockerfile",
-		"args": {
-			"NODE_VERSION": "lts/*"
-		}
-	},
-
-	// Configure tool-specific properties.
-	"customizations": {
-		// Configure properties specific to VS Code.
-		"vscode": {
-			// Set *default* container specific settings.json values on container create.
-			"settings": { 
-				"python.defaultInterpreterPath": "/opt/conda/bin/python",
-				"python.linting.enabled": true,
-				"python.linting.pylintEnabled": true,
-				"python.formatting.autopep8Path": "/opt/conda/bin/autopep8",
-				"python.formatting.yapfPath": "/opt/conda/bin/yapf",
-				"python.linting.flake8Path": "/opt/conda/bin/flake8",
-				"python.linting.pycodestylePath": "/opt/conda/bin/pycodestyle",
-				"python.linting.pydocstylePath": "/opt/conda/bin/pydocstyle",
-				"python.linting.pylintPath": "/opt/conda/bin/pylint"
-			},
-			
-			// Add the IDs of extensions you want installed when the container is created.
-			"extensions": [
-				"ms-python.python",
-				"ms-python.vscode-pylance",
-				"njpwerner.autodocstring",
-				"ms-toolsai.jupyter",
-				"donjayamanne.python-environment-manager",
-				"ms-azuretools.vscode-docker"
-			]
-		}
+	// Conda requires lots of memory to resolve our environment
+	"hostRequirements": {
+		"memory": "8gb"
 	},
 
-	// Use 'forwardPorts' to make a list of ports inside the container available locally.
-	// "forwardPorts": [],
+	// Use 'postCreateCommand' to run commands after the container is created
+	"postCreateCommand": "conda env create -f environment.yml",
 
-	// Use 'postCreateCommand' to run commands after the container is created.
-	"postCreateCommand": "conda init",
-
-	// Comment out to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
-	"remoteUser": "vscode",
+	// Features to add to the dev container. More info: https://containers.dev/features
 	"features": {
 		"ghcr.io/devcontainers/features/rust:1": {
 			"version": "latest"
```

Comment:
Thanks, this looks much more minimal :).  Does this pull in conda?  NumPy itself shouldn't need rust.

There will be some additional setup necessary probably (such as submodule initialization).
It would be nice to do some of that automatically, it could even start installing numpy and running `make html` in the docs directory.  The gitpod setup does that I think (and maybe even caches the container after it).

I also liked the idea of printing a brief info maybe just with a link to further docs, but those docs don't yet exist probably, so that has no priority.

Reply:
Cool, removing Rust will reduce creation time! Yep, Conda is included in the [Dev Containers universal image](https://github.com/devcontainers/images/tree/main/src/universal). The tool (and Codespaces) use that image by default, but we can specify it with `”image”` for clarity.

If there’s lots of setup work to be done, we can run a `setup.sh` in `”postCreateCommand”` that is responsible for doing all of this.

Some follow up questions:
- Do you use Conda for all numpy dev work?
- Do you know how to make VS Code auto-activate a Conda env?

Reply:
I don't, but it is a typical setup and probably the easiest to standardize.  You could also use pip and/or system libs (but some pip will be needed).

I don't know about VS code and conda envs unfortunately.

Reply:
OK, still slow to run all the conda setup (slow enough that I forgot about the browser tab ;)).  But things seem to work.  After dropping in, it would be nice to:
1. Make sure `main` tracks `upstream` (maybe this would already be true if run on the main branch though).
2. Maybe run `git pull upstream` and `git submodule update --init` (necessary for compiling numpy)

With that setup:
* `python3 runtests.py` works and succeeds.
* `python3 runtests.py --shell` then `cd docs` and `make html` seems to also work fine  (maybe not the most standard workflow, but works).
  (but as mentioned on slack, I am not sure how to open the static html, since this won't work `python3 -m webbrowser build/html/index.html`)

Which makes the environment slow to start up, but usable.

@jungaretti besides removing rust, are you planning on doing more here?

It seems good to just get this in and make improvements kater.  This definitely makes it more usable and much more lean.

Reply:
Yea, the Conda env creation is really slow. Unfortunately, I think that is a limitation with Conda instead of Dev Containers/Codespaces.

Can you elaborate on your `upstream`/`main` comments? Those are distinct locations when working with repo forks.

1. Once this merges and you make a codespace for numpy/numpy (instead of jungaretti/numpy), `main` and `upstream` will be one and the same.
2. We can definitely add those steps.

I'm not planning to make any more changes. I'll clean up Rust, polish this PR, and mark it as ready for review!

Reply:
ah, most of our docs use `upstream` as the NumPy repo, and `origin` as the users repo (where they have to make branches for a PR).  So setting this up automatically the same way could be nice.
If users do `git pull` on main, it should pull from `upstream` (i.e. `numpy/numpy`) and not the users fork.